ha/core/event_manager/event_manager.py

```diff
@@ -71,28 +192,78 @@ def _validate_events(self, events: list) -> None:
                 raise InvalidEvent(f"Invalid event: {event}, not part of HA event list.")
         Log.debug(f"event: {event} is valid for subscription request")
 
-    def subscribe(self, component: str, events: list) -> str:
-        """
-        Register events for the notification. It maintains list of events registered by the components.
-
-        Args:
-            component (str): Component Name
-            events (list): Events.
-        """
-        pass
+    def _delete_component_key(self, component: str, resource_type: str, states: list = None) -> None:
```

Comment:
delete_component_key
_delete_event_key
same as above

Reply:
Same as above.

Reply:
There is some code duplication but will handle in refactor